xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -151,6 +156,7 @@
       Strings.isNullOrEmpty(System.getenv("GRPC_XDS_EXPERIMENTAL_RBAC"))
           || Boolean.parseBoolean(System.getenv("GRPC_XDS_EXPERIMENTAL_RBAC"));
 
+  private static final long METADATA_SYNC_TIMEOUT_SEC = 1;
```

Comment:
I'm feeling 1 sec is too short in case the machine is slow and busy with GC. I would wait about 20 sec, and in normal case it wouldn't take more time, and if bad things still happen, in any case it would fail.

Reply:
Yea, this constant was completely arbitrary guess. I still think 1 sec might be reasonable though:

> I'm feeling 1 sec is too short in case the machine is slow and busy with GC.

Do you have an operation in mind that would hold this lock for 20 seconds?

> and if bad things still happen, in any case it would fail

In case of CSDS, I'd rather fail if it doesn't complete in 1 sec than accidentally hold xds synchronisation lock for more than 1 second. I wonder if I should just make the wait time a parameter.

Reply:
> Do you have an operation in mind that would hold this lock for 20 seconds?

For example, a GC pause can be very long. Although that is rare, if that happens, the application will fail if the timeout is just 1 sec. And you know normally the lock won't be hold very long, so set a large timeout doesn't hurt anything.

Reply:
We've seen very high times randomly through all code. Java starts with a slow interpreter and class loading does disk I/O. We've routinely seen trivial code take longer than a 2 seconds on a brand-new JVM if run many times (like in our CI). Then you also have cases where the CPU is overloaded. (And GC, etc)